src/App.jsx

```diff
@@ -1,105 +1,70 @@
-import './App.css';
-import {Link} from 'react-router-dom';
-import axios from 'axios';
-import { useEffect, useState } from 'react';
+import "./App.css";
+import { fetchProducts } from "./api";
+import { useEffect, useState } from "react";
 
-import Header from './component/Header';
-import BestProducts from './component/BestProducts';
-import ProductList from './component/ProductList';
-import Pagination from './component/Pagination';
-import SortMenu from './component/SortMenu';
+import Header from "./component/Header";
+import { Routes, Route } from "react-router-dom";
+import Additem from "./page/additem";
+import MainContent from "./page/MainContent";
 
 function App() {
-  const [bestProducts, setBestProducts] = useState([]); 
-  const [isOpen, setOpen] = useState(false);
-  const [sortbtntext, setSortbtntext] = useState('최신순');
+  const [bestProducts, setBestProducts] = useState([]);
   const [sortedProducts, setSortedProducts] = useState([]);
-  const pagebuttons = [1, 2, 3, 4, 5];
+
+  const [totalpage, setTotalpage] = useState(0);
   const [activePage, setActivePage] = useState(1);
-  const [sortOrder, setSortOrder] = useState('recent'); 
+  const [sortOrder, setSortOrder] = useState("recent");
 
- 
   useEffect(() => {
+    const mediaQuery = window.matchMedia("(max-width: 768px)");
     const fetchBestProducts = async () => {
+      const pageSize = mediaQuery.matches ? 2 : 4;
       try {
-        const response = await axios.get(`https://panda-market-api.vercel.app/products?page=1&orderBy=favorite`);
-        const bestProducts = response.data.list.slice(0, 4); 
-        setBestProducts(bestProducts);
+        const data = await fetchProducts(1, "favorite", pageSize);
+        const totalpage = Math.ceil(data.totalCount / 10);
+        setTotalpage(totalpage);
+        setBestProducts(data.list);
```

Comment:
P3:
아래 코드도 동일동작합니다.

```suggestion
        const {totalCount, list} = await fetchProducts(1, "favorite", pageSize);
        const totalpage = Math.ceil(totalCount / 10);
        setTotalpage(totalpage);
        setBestProducts(list);
```